### ISSUE TITLE
[Bug]: fatal web ,error: deadlock detected

### ISSUE THREAD

**Issue by msy999** [nextcloud/server#31635]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

ERROR: deadlock detected
DETAIL: Process 154 waits for ShareLock on transaction 4402; blocked by process 159.
Process 159 waits for ShareLock on transaction 4384; blocked by process 154.
HINT: See server log for query details.
CONTEXT: while rechecking updated tuple (5,29) in relation "oc_filecache"

### Steps to reproduce

1. server23.02 , mariadb 10.5 ,redis-alpine
2. If you install a new one and upload a lot of data at the same time, the above error message will appear.

### Expected behavior

ERROR: deadlock detected
DETAIL: Process 154 waits for ShareLock on transaction 4402; blocked by process 159.
Process 159 waits for ShareLock on transaction 4384; blocked by process 154.
HINT: See server log for query details.
CONTEXT: while rechecking updated tuple (5,29) in relation "oc_filecache"

### Installation method

Official Docker image

### Operating system

Other

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
*
```

### List of activated Apps

```shell
*
```

### Nextcloud Signing status

```shell
*
```

### Nextcloud Logs

```shell
*
```

### Additional info

*

**Comment by szaimen** [comment]
Hi, please update to 24.0.9 or better 25.0.3 and report back if it fixes the issue. Thank you!

My goal is to add a label like e.g. 25-feedback to this ticket of an up-to-date major Nextcloud version where the bug could be reproduced. However this is not going to work without your help. So thanks for all your effort!

If you don't manage to reproduce the issue in time and the issue gets closed but you can reproduce the issue afterwards, feel free to create a new bug report with up-to-date information by following this link: https://github.com/nextcloud/server/issues/new?assignees=&labels=bug%2C0.+Needs+triage&template=BUG_REPORT.yml&title=%5BBug%5D%3A+

**Comment by obel1x** [comment]
@msy999
Looks like its related to the same behaviour as #6899. See my proposal to reduce deadlocks by changing indices on table oc_filecache. Can you check https://github.com/nextcloud/server/issues/6899#issuecomment-1377954159 and leave a thumb up there if it works?

**Comment by szaimen** [comment]
Duplicate of https://github.com/nextcloud/server/issues/6899 then
